PyInstaller/hooks/rthooks/pyi_rth_inspect.py

```diff
@@ -20,7 +20,7 @@
 # filenames obtained from object (e.g., inspect stack-frames). See #5963.
 def _pyi_getsourcefile(object):
     filename = inspect.getfile(object)
-    if not os.path.isabs(filename):
+    if hasattr(sys.modules['__main__'], "__file__") and not os.path.isabs(filename):
```

Comment:
I don't think this is quite right; this if-block has two sub-blocks, and only the first subblock depends on `sys.modules['__main__'].__file__`.

The change should be:
```python
...
if not os.path.isabs(filename):
    # Check if given filename matches the basename of __main__'s __file__.
    main_file = getattr(sys.modules['__main__'], '__file__')
    if main_file and filename == os.path.basename(main_file):
...
```

Reply:
I'm going to check if your proposal also solves the issue.

Reply:
Updated.